### PR TITLE
dts: Standardize string token names for arrays

### PR DESCRIPTION
**Pull request by galak**
Use the same postfixes for string arrays as we use for strings:

	_TOKEN -> _STRING_TOKEN
	_UPPER_TOKEN -> _STRING_UPPER_TOKEN

Signed-off-by: Kumar Gala <galak@kernel.org>

### CODE REVIEW

**Comment by galak** [comment]
NOTE: We should not have exposed the `*TOKEN` in `soc/arm/xilinx_zynq7000/common/pinctrl_soc.h`.  There should have been an API defined to use this (https://github.com/zephyrproject-rtos/zephyr/pull/46249)

**Review by henrikbrixandersen — DISMISSED** [review]

**Comment by galak** [comment]
Will merge https://github.com/zephyrproject-rtos/zephyr/pull/46249 first.

**Review by henrikbrixandersen — DISMISSED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

File: scripts/dts/gen_defines.py [hunk 1]
```diff
@@ -623,8 +623,6 @@ def write_vanilla_props(node):
                 if isinstance(subval, str):
                     macro2val[macro + f"_IDX_{i}"] = quote_str(subval)
                     subval_as_token = edtlib.str_as_token(subval)
-                    macro2val[macro + f"_IDX_{i}_TOKEN"] = subval_as_token
-                    macro2val[macro + f"_IDX_{i}_UPPER_TOKEN"] = subval_as_token.upper()
```
Comment (mbolivar-nordic) [inline scripts/dts/gen_defines.py:627]: Needs macros.bnf update

**Review by mbolivar-nordic — CHANGES_REQUESTED** [review]

File: scripts/dts/gen_defines.py (hunk 1 above)
Comment (galak) [reply scripts/dts/gen_defines.py:627]: Can we do that as a follow PR, since none of the string TOKENs are documented in `macros.bnf`

**Review by mbolivar-nordic — APPROVED** [review]

File: scripts/dts/gen_defines.py (hunk 1 above)
Comment (mbolivar-nordic) [reply scripts/dts/gen_defines.py:627]: > Can we do that as a follow PR

OK

**Pull request merged**